api/v1beta1/clusterclass_types.go

```diff
@@ -518,7 +521,125 @@ type JSONSchemaProps struct {
 	// NOTE: Can be set for all types.
 	// +optional
 	Default *apiextensionsv1.JSON `json:"default,omitempty"`
-}
+
+	// XValidations describes a list of validation rules written in the CEL expression language.
+	// +optional
+	XValidations ValidationRules `json:"x-kubernetes-validations,omitempty"`
```

Comment:
Hyphens in field names is not Kuberentes-like. This field should be `xValidations` or `xKubernetesValidations`. Typically the field name and the json are the same, one being PascalCase and one being camelCase.

Does this need to be an exact copy of the json schema props from the CRD extension API?

Reply:
Up until now we tried to keep the API compatible with upstream CRDs (for the fields that we implement there) so that folks that are using ClusterClass can basically use the same as for CRDs

Reply:
Makes sense in this case to keep the compatible schema then

Reply:
Thanks for chipping in @sbueringer! Yes we are trying to keep the API as close to CRD CEL API defs as possible. The API had been previously reviewed and approved before I helped out with the implementation, which is what I was hoping for feedback on at this point, but if we do need to revisit the API then I'm happy to do so, but still remembering the desire to stick to CRD CEL validation semantics.